### PR TITLE
SAK-30354 Change default portlet property prefix from javax.portlet

### PR DESCRIPTION
**Pull request by jonespm**

### CODE REVIEW

File: portal/portal-render-impl/impl/src/java/org/sakaiproject/portal/render/portlet/services/SakaiOptionalPortletContainerServices.java [hunk 1]
```diff
@@ -350,8 +350,12 @@ public SakaiPortletPreferencesService()
 				for (Enumeration e = props.propertyNames(); e.hasMoreElements();)
 				{
 					String propertyName = (String) e.nextElement();
-					// System.out.println("Property name = "+propertyName);
-					if (propertyName != null && propertyName.startsWith("javax.portlet:")
+					if (M_log.isDebugEnabled())
```
Comment (steveswinsburg) [inline portal/portal-render-impl/impl/src/java/org/sakaiproject/portal/render/portlet/services/SakaiOptionalPortletContainerServices.java]: isDebugEnabled() not necessary if you are just concatenating strings.

**Comment by csev** [comment]
I am completely misunderstanding what you are doing here.   Just changing the prefix will break all the existing JSR-168 tools that have been storing and retrieving properties for the past five years.

**Comment by csev** [comment]
I just looked more closely at the JIRA.  I see you are proposing a database conversion.  

I don't think that is enough.  I think that the code needs to look for both the old and new prefix and then patch / adjust / detect on-the-fly.   The code should work in a way that the database conversion is unnecessary but we can also convert the database at upgrade time.

**Comment by jonespm** [comment]
Hi @csev 

There is should be no database conversion needed the way this is done.

It changes the lookup from
-					if (propertyName != null && propertyName.startsWith("javax.portlet:")
+					if (propertyName != null && (propertyName.startsWith("javax.portlet"))

So it just sees if it starts with the "javax.portlet" (not considering what symbolic character is used) but the length still needs to match up. In testing this should ensure all old properties are still able to be processed, just new ones are added with a -.

There will be a mix of javax.portlet: and javax.portlet- in the database, and a conversion probably would be "nice" to clean it up, but shouldn't be 100% necessary.

**Pull request merged**